core/templates/pages/topics-and-skills-dashboard-page/delete-skill-modal.controller.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { UpgradedServices } from 'services/UpgradedServices';
+// ^^^ This block is to be removed.
+
+describe('Delete Skill Modal Controller', function() {
+  var $scope = null;
+  var $rootScope = null;
+  var $q = null;
+  var $uibModalInstance = null;
+  var TopicsAndSkillsDashboardBackendApiService = null;
+  var skillId = 'skillId1';
+  beforeEach(angular.mock.module('oppia'));
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  describe('when skill has no assigned topics', function() {
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      $rootScope = $injector.get('$rootScope');
+      $q = $injector.get('$q');
+      TopicsAndSkillsDashboardBackendApiService = (
+        $injector.get('TopicsAndSkillsDashboardBackendApiService'));
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+      var MockTopicsAndSkillsDashboardBackendApiService = {
+        fetchAssignedSkillData: () => {
+          var deferred = $q.defer();
+          deferred.resolve({
+            assigned_topics: {}
+          });
+          return deferred.promise;
+        }
+      };
+      $scope = $rootScope.$new();
+      $controller('DeleteSkillModalController', {
+        $scope: $scope,
+        TopicsAndSkillsDashboardBackendApiService:
+        MockTopicsAndSkillsDashboardBackendApiService,
+        $uibModalInstance: $uibModalInstance,
+        skillId
+      });
+    }));
+
+    it('should initialize the correct value', function() {
+      $scope.init();
+      $rootScope.$apply();
+      expect($scope.assignedTopics).toEqual({});
+      expect($scope.assignedTopicsAreFetched).toEqual(true);
+      expect($scope.showAssignedTopics()).toEqual(false);
+    });
+  });
+
+  describe('when skill has assigned topics', function() {
+    var topicIdDict = {topic1: {id: 'topicId1'}};
+    beforeEach(angular.mock.inject(function($injector, $controller) {
+      $rootScope = $injector.get('$rootScope');
+      $q = $injector.get('$q');
+      TopicsAndSkillsDashboardBackendApiService = (
+        $injector.get('TopicsAndSkillsDashboardBackendApiService'));
+
+      $uibModalInstance = jasmine.createSpyObj(
+        '$uibModalInstance', ['close', 'dismiss']);
+      var MockTopicsAndSkillsDashboardBackendApiService = {
+        fetchAssignedSkillData: () => {
+          var deferred = $q.defer();
+          deferred.resolve({
+            assigned_topics: topicIdDict
+          });
+          return deferred.promise;
+        }
+      };
+      $scope = $rootScope.$new();
+      $controller('DeleteSkillModalController', {
+        $scope: $scope,
+        TopicsAndSkillsDashboardBackendApiService:
+        MockTopicsAndSkillsDashboardBackendApiService,
+        $uibModalInstance: $uibModalInstance,
+        skillId
```

Comment:
ditto: key for `skillId`

Reply:
Ah, Used the javascript shorthard operator, which can be used if the key and the variable for value have the same name.
But made this simple, so that no confusion is introduced. 
Done.

Reply:
I see. Thanks!